controllers/ProposalController.php

```diff
@@ -1144,4 +1167,125 @@ private function saveEditedReview(\app\models\Review $review, $reviewStatus)
         return $review;
     }
 
+    /**
+     * Check if user can publish Proposal
+     *
+     * @param Proposal $proposal
+     * @return bool
+     */
+    private function canPublishProposal(Proposal $proposal)
+    {
+        $currentUser = MainController::getCurrentUser();
+
+        if (($currentUser->role === User::USER_ROLE_PUBLISHER || $currentUser->role === User::USER_ROLE_ADMIN)
+            && ($proposal->submitter_id !== $currentUser->id) && $proposal->status === \app\models\Proposal::STATUS_PENDING) {
+            return true;
+        }
```

Comment:
You could use `$currentUser->hasRole(User::USER_ROLE_PUBLISHER)` instead of `$currentUser->role === User::USER_ROLE_PUBLISHER || $currentUser->role === User::USER_ROLE_ADMIN`

Reply:
Shame on me ! I totally forgot this method. I change that